src/waveform/renderers/waveformmarkproperties.h

```diff
@@ -0,0 +1,27 @@
+#ifndef WAVEFORMMARKPROPERTIES_H
+#define WAVEFORMMARKPROPERTIES_H
+
+#include <QColor>
+#include <QDomNode>
+
+class SkinContext;
+class WaveformSignalColors;
+
+class WaveformMarkProperties {
+  public:
+    WaveformMarkProperties();
+    WaveformMarkProperties(const QDomNode& node,
+                           const SkinContext& context,
+                           const WaveformSignalColors& signalColors);
+    virtual ~WaveformMarkProperties();
```

Comment:
...then we can remove the keyword final and add a virtual destructor.

But why should we introduce additional overhead (like a vtable) for a class 
without any virtual functions? The keyword 'final' ensures that we have to 
think about this again if we decide that we actually need subclasses. Don't 
over-engineer, keep it simple (but not stupid). With C++11 at hand it would 
be stupid to omit the keyword 'final' and go without a virtual destructor.

On 21.08.2016 14:18, Ferran Pujol Camins wrote:
>
> In src/waveform/renderers/waveformmarkproperties.h 
> <https://github.com/mixxxdj/mixxx/pull/992#discussion_r75594384>:
>
> > +#ifndef WAVEFORMMARKPROPERTIES_H
> > +#define WAVEFORMMARKPROPERTIES_H
> > +
> > +#include <QColor>
> > +#include <QDomNode>
> > +
> > +class SkinContext;
> > +class WaveformSignalColors;
> > +
> > +class WaveformMarkProperties {
> > +  public:
> > +    WaveformMarkProperties();
> > +    WaveformMarkProperties(const QDomNode& node,
> > +                           const SkinContext& context,
> > +                           const WaveformSignalColors& signalColors);
> > +    virtual ~WaveformMarkProperties();
>
> What if in the future we want to have specialized WaveFormMarks? We will 
> probably subclass WaveFormMarks and also subclass WaveformMarkProperties 
> to add new fields.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/992/files/2f8863b7b8266565a97581fd38cfa6e87ab0ac36#r75594384>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-anTInxctyi7QrSh-dmzxwMeNK82-ks5qiEIogaJpZM4Jj7vm>.
>